packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -3209,95 +3210,776 @@ describe('NetworkController', () => {
+          );
+        });
+
+        it('does not set the new network to active by default', async () => {
+          await withController(
+            { infuraProjectId: 'some-infura-project-id' },
+            async ({ controller }) => {
+              uuidV4Mock.mockReturnValue('AAAA-AAAA-AAAA-AAAA');
+              const builtInNetworkProvider = buildFakeProvider([
+                {
+                  request: {
+                    method: 'test_method',
+                    params: [],
+                  },
+                  response: {
+                    result: 'test response from built-in network',
+                  },
+                },
+              ]);
+              const builtInNetworkClient = buildFakeClient(
+                builtInNetworkProvider,
+              );
+              const newCustomNetworkProvider = buildFakeProvider([
+                {
+                  request: {
+                    method: 'test_method',
+                    params: [],
+                  },
+                  response: {
+                    result: 'test response from custom network',
+                  },
+                },
+              ]);
+              const newCustomNetworkClient = buildFakeClient(
+                newCustomNetworkProvider,
+              );
+              mockCreateNetworkClient()
+                .calledWith({
+                  network: NetworkType.mainnet,
+                  infuraProjectId: 'some-infura-project-id',
+                  type: NetworkClientType.Infura,
+                })
+                .mockReturnValue(builtInNetworkClient)
+                .calledWith({
+                  chainId: toHex(111),
+                  rpcUrl: 'https://test.network',
+                  type: NetworkClientType.Custom,
+                })
+                .mockReturnValue(newCustomNetworkClient);
+              // Will use mainnet by default
+              await controller.initializeProvider();
+
+              await controller.upsertNetworkConfiguration(
+                {
+                  rpcUrl: 'https://test.network',
+                  chainId: toHex(111),
+                  ticker: 'TICKER',
+                },
+                {
+                  setActive: false,
+                  referrer: 'https://test-dapp.com',
+                  source: 'dapp',
+                },
+              );
+
+              const { provider } = controller.getProviderAndBlockTracker();
+              assert(provider, 'Provider is not set');
+              const { result } = await promisify(provider.sendAsync).call(
+                provider,
+                {
+                  id: 1,
+                  jsonrpc: '2.0',
+                  method: 'test_method',
+                  params: [],
+                },
+              );
+              expect(result).toBe('test response from built-in network');
+            },
+          );
+        });
+      });
+
+      describe('if the setActive option is true', () => {
```

Comment:
Hmm, do you mean have a test which confirms that the provider has been set? If so, I left this out because the `refreshNetwork` tests effectively do this already: https://github.com/MetaMask/core/blob/rewrite-network-configuration-tests/packages/network-controller/tests/NetworkController.test.ts#L5767.

Reply:
Thanks, makes sense